pkg/cmd/step/step_custom_pipeline.go

```diff
@@ -106,6 +106,11 @@ func (o *StepCustomPipelineOptions) Run() error {
 		return err
 	}
 
+	gitKind, err := o.GitServerKind(gitInfo)
```

Comment:
OOI, how does this work under the hood. Stepping into the code it first seems to look at the URL host to see whether it is one of the known git hosting sites, but then it continues to do `GetServiceKindFromSecrets` where I start wondering.

Reply:
The important bit at this point is https://github.com/jenkins-x/jx/blob/master/pkg/kube/git_services.go#L99 and below - checking `jx-auth-config` to see if we’ve got creds for the particular host and if so, what provider kind it is. It does first check if it’s a known hosting provider, and after the auth config check falls back further to the secrets (no longer relevant in modern installs thanks to the auth config) and finally a super old legacy check for a CRD.

Reply:
> to see if we’ve got creds for the particular host and if so, what provider kind it is.

and where does this kind come from? It is not determined in some way, it must be provided, right?

Reply:
In the end, it comes from when git creds are defined and configuration in `jx-requirements.yml` driving that.

Reply:
> In the end, it comes from when git creds are defined and configuration in jx-requirements.yml driving that.

would it not more direct to go via the requirements then?  That's a different discussion really, just wondering. It seems a bit odd to me to go all the way to the secrets and whatnot, but still ending up with some indirect information.

Reply:
I’ve debated going that route, but I think it is possible to add more secrets for other providers outside of requirements, which is...a thing. Next time I dive through this whole space for anything more than the most trivial of fixes to the “supported” providers (GH, GL, BBS), I.e., when I get around to adding BitBucket Cloud support, I’m going to first dedicate a few days to mapping out all the permutations of this junk to handle it properly.

Reply:
> when I get around to adding BitBucket Cloud support, I’m going to first dedicate a few days to mapping out all the permutations of this junk to handle it properly.

Maybe at least create a place holder issue!?

Reply:
https://github.com/jenkins-x/jx/issues/7201